src/common/utils/compare.ts

```diff
@@ -10,30 +17,226 @@ export interface RankedLocationSummary extends SummaryForCompare {
   rank: number;
 }
 
+export const orderedMetrics = [
+  Metric.CASE_DENSITY,
+  Metric.CASE_GROWTH_RATE,
+  Metric.POSITIVE_TESTS,
+  Metric.HOSPITAL_USAGE,
+  Metric.CONTACT_TRACING,
+];
+
 const locations: any = getLocationNames();
 
-export function getStatesArr(): SummaryForCompare[] {
+function getLocationObj(locationInfo: Location) {
+  if (locationInfo.full_fips_code) {
+    return {
+      locationInfo: locationInfo,
+      metricsInfo:
+        locationInfo.full_fips_code &&
+        countySummary(locationInfo.full_fips_code),
+    };
+  } else {
+    return {
+      locationInfo: locationInfo,
+      metricsInfo: stateSummary(locationInfo.state_code),
+    };
+  }
+}
+
+function isCounty(location: Location) {
+  return location.county;
+}
+
+function isState(location: Location) {
+  return !location.county;
+}
+
+function isCountyOfState(location: Location, stateId: string) {
+  return location.state_code === stateId;
+}
+
+function isMetroCounty(location: Location) {
+  return location.full_fips_code && getCountyMsaCode(location.full_fips_code);
+}
+
+function isNonMetroCounty(location: Location) {
+  return location.full_fips_code && !getCountyMsaCode(location.full_fips_code);
+}
```

Comment:
Consider using `isMetroCounty` here since one is the opposite of the other

```ts
const isNonMetroCounty = (location: Location) => !isMetroCounty(location)
```